src/scripts/main.js

```diff
@@ -1,3 +1,216 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+const tableHeaders = document.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+
+let isAsc = true;
+let defaultIndex = -1;
+
+tableHeaders.addEventListener('click', (e) => {
+  const tableRows = tableBody.querySelectorAll('tr');
+
+  sortData([...tableRows], e.target.cellIndex);
+});
+
+const sortData = (arr, index) => {
+  isAsc = index !== defaultIndex || !isAsc;
+  defaultIndex = index;
+
+  arr.sort((a, b) => {
+    let prev = a.children[index].innerText.trim();
+    let next = b.children[index].innerText.trim();
+
+    if (!isAsc) {
+      [prev, next] = [next, prev];
+    }
+
+    if (getNumber(prev)) {
+      return getNumber(prev) - getNumber(next);
+    } else {
+      return prev.localeCompare(next);
+    }
+  });
+
+  for (const x of arr) {
+    tableBody.append(x);
+  }
+};
+
+const getNumber = (data) => data.replace(/\D/g, '');
+
+tableBody.addEventListener('click', e => {
+  Array.from(e.target.parentElement.parentElement.children).forEach(el => {
```

Comment:
It's not clear what exactly is `e.target.parentElement.parentElement.children` use `tableBody.children` instead